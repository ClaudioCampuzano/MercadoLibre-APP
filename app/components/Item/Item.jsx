import React from 'react';
import Link from 'next/link';
import { formatCurrency } from '@/utils/formatCurrency';
import Image from 'next/image';
import styles from './Item.module.scss';

const Item = ({ data }) => {
  const { title, picture, condition, price } = data;
  return (
    <div className={styles.container}>
      <Link href={`/items/${data.id}`} className={styles.link}>
        <article className={styles.article}>
          <Image src={picture} alt="article" width={150} height={150} />
          <section>
            <div className={styles.price}>
              <h2>{formatCurrency(price)}</h2>
              {condition === 'new' && 'ðŸŸ¢'}
            </div>
            <h3>{title}</h3>
          </section>
        </article>
      </Link>
    </div>
  );
};

export { Item };
