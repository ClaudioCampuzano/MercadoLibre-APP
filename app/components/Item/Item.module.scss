@import '@/styles/constans.module.scss';

.container {
  background-color: $whiteColor;
  margin-top: 0.1rem;
  padding: 15px;
  width: 80%;
}

.link {
  text-decoration: none;
  color: $primaryTextColor;
}

.article {
  display: flex;
}

.price {
  display: flex;
  align-items: center;
}
