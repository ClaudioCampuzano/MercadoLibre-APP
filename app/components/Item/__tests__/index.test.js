import React from 'react';

import { render } from '@testing-library/react';

import { Item } from '../Item';

jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));

jest.mock('@/utils/formatCurrency', () => ({
  formatCurrency: () => '$456789',
}));

const data = {
  title: 'ipsum',
  picture: 'htpp://blabla.com',
  condition: 'new',
  price: { amount: 15121, currency: 'CLP' },
};

const componentRender = (props = { data }) => {
  return render(<Item {...props} />);
};

describe('Item', () => {
  it('renders', () => {
    const { container } = componentRender();
    expect(container).toMatchSnapshot();
  });
});
