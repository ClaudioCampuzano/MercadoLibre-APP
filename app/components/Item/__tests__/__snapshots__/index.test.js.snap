// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item renders 1`] = `
<div>
  <div
    class="container"
  >
    <a
      class="link"
      href="/items/undefined"
    >
      <article
        class="article"
      >
        <img
          alt="article"
          data-nimg="1"
          decoding="async"
          height="150"
          loading="lazy"
          src="/_next/image?url=htpp%3A%2F%2Fblabla.com&w=384&q=75"
          srcset="/_next/image?url=htpp%3A%2F%2Fblabla.com&w=256&q=75 1x, /_next/image?url=htpp%3A%2F%2Fblabla.com&w=384&q=75 2x"
          style="color: transparent;"
          width="150"
        />
        <section>
          <div
            class="price"
          >
            <h2>
              $456789
            </h2>
            ðŸŸ¢
          </div>
          <h3>
            ipsum
          </h3>
        </section>
      </article>
    </a>
  </div>
</div>
`;
